### PR TITLE
feat: add ASCIIToSVG (a2s binary) to container image.

### LINKED ISSUE #371: Add ASCIIToSVG support?
Hello

Thanks for the work that goes into Asciidoc Docker Container.

We are using [ASCIIToSVG](https://docs.asciidoctor.org/diagram-extension/latest/diagram_types/a2s/).  Would you consider a PR that includes the building of `a2s` in the Dockerfile so the binary is available within the container.  I'd be happy to put up a PR. 

Thank you.

### PR DESCRIPTION
**Pull request by k-wall**
Fix #371.

### CODE REVIEW

File: Dockerfile
Comment (k-wall) [inline Dockerfile:47]: asciitosvg repo isn't tagged. I chose to base the Dockerfile on main's current SHA so that any updates are made deliberately  and container builds have repeatable content.

**Comment by k-wall** [comment]
Test cases added now too:

```
kwall@Oslo docker-asciidoctor % export IMAGE_NAME=docker.io/asciidoctor/docker-asciidoctor
kwall@Oslo docker-asciidoctor % bats tests
 ✗ We can generate an HTML document with a diagram with asciidoctor-diagram as backend
   (in test file tests/asciidoctor.bats, line 154)
     `[ "$(echo ${output} | grep -c -i error)" -eq 0 ]' failed
   -- Output of command:
   Trying to pull docker.io/asciidoctor/docker-asciidoctor:latest...
   Getting image source signatures
   Copying blob sha256:56782d0530714adc62ae576e85116d4dda40912cfe97f62e81c2a2627d79da37
   Copying blob sha256:31e352740f534f9ad170f75378a84fe453d6156e40700b882d737a8f4a6988a3
   Copying blob sha256:ca25e8f85fd11c124a1401de1836343fea29f976924d1517201a4302f7932c14
   Copying blob sha256:8d9db96ef65be24a4fd269bc9cce32c3b051e75c86a05e0363f84803f7ffde02
   Copying blob sha256:aae430c7a9ece80aa0cdb3fbf9d2f60df80458d4f55e40577bb174945888e9b7
   Copying blob sha256:051b8c71077c967b14c39eac415a72e9ec18d2b81a8e546897f793651be9557a
   Copying blob sha256:a488ac73979d6f2b11679a5094601ed1e21fa06ccc4a2e599fd507900be2f96c
   Copying blob sha256:565ef0a8bb3b7a1103e77c916ed9ce2f2d3659b2a4179e66ccd95f1c3a8607ec
   Copying config sha256:7b6c7a7174cf3b13e4cea5280e15451f75b71a7aed9d52a93abb1b69038152fa
   Writing manifest to image destination
   Storing signatures
   asciidoctor: ERROR: sample-with-diagram.adoc: line 57: Failed to generate image: Could not find the 'a2s' executable in PATH; add it to the PATH or specify its location using the 'a2s' document attribute
   --


1 test, 1 failure
kwall@Oslo docker-asciidoctor % export IMAGE_NAME=quay.io/k_wall/docker-asciidoctor:latest
kwall@Oslo docker-asciidoctor %
kwall@Oslo docker-asciidoctor %
kwall@Oslo docker-asciidoctor % bats tests
 ✓ We can generate an HTML document with a diagram with asciidoctor-diagram as backend

1 test, 0 failures
```

**Comment by k-wall** [comment]
Push an update in response to this comment:

In https://github.com/k-wall/docker-asciidoctor/commit/d14a84deedfbf49ab83e61dea3dfa3ed465a2ab0#r120679120 @barthel said:
> Use the same Alpine version as the Asciidoctor image here would be more safe.
> See: https://github.com/barthel/docker-asciidoctor/blob/3f41f7f1d10a9a6face36ee436aefff9781f35eb/Dockerfile#L5

**Comment by k-wall** [comment]
Any feedback on this PR would be appreciated.

File: Dockerfile
Comment (dduportal) [inline Dockerfile:2]: This line change is unrelated to the pull request. But since we don't track this version with an `updatecli` manifest, no problem to change it like this for homogeneity

File: Dockerfile
Comment (dduportal) [reply Dockerfile:47]:
```suggestion
# Expects a git reference as there are no tags in the A2S repository
ARG A2S_VERSION=ca82a5c
```

Thanks for the explanation, it make sense. WDYt about a comment here until we can track the reference with `updatecli` ?

File: Dockerfile
Comment (dduportal) [inline Dockerfile:48]:
```suggestion
RUN GOBIN=/app go install github.com/asciitosvg/asciitosvg/cmd/a2s@"${A2S_VERSION}"
```

As the build arg can be specified with unexpected values, let's apply shellcheck good practises

File: tests/fixtures/sample-with-diagram.adoc
Comment (dduportal) [inline tests/fixtures/sample-with-diagram.adoc:55]: Blocking: since you're adding this sample, could you add a corresponding test like other diagram systems?

**Review by dduportal — CHANGES_REQUESTED** [review]
Hello, thanks for the contribution.

- It's a legit change request
- The added size is minimal (due to your proper usage of multi stage build)
- Versions are fixed to ensure constant behavior over time

I only have one blocking point: can you add a test with the code sampel you provided? You added a test checking for the presence of the binary in the `PATH`  which is a good start, but test harness should run a full execution with your sample to really cover more cases.


Provided a few nitpicks suggestions or comments (not blocking the merge).

Please note that, once this PR is merged, there would be a need to subsequent distinct PR  to add the updatecli manifest tracking the git reference of A2S for future updates: not mandatory for you to do it, but if you are interested let me know to avoid stepping on your toes

File: tests/fixtures/sample-with-diagram.adoc
Comment (k-wall) [reply tests/fixtures/sample-with-diagram.adoc:55]: I'm not quite sure what you are asking for with this comment.   With the change made to `sample-with-diagram.adoc`, a2s is being exercised by `@test "We can generate an HTML document with a diagram with asciidoctor-diagram as backend"`.  I verified that the new test data causes a [fail](https://github.com/asciidoctor/docker-asciidoctor/pull/372#issuecomment-1613446738) against the current image.

As far as I can see, that testing approach is  consistent with the tests for the other diagram systems.  Could you be more specific?

Comment (dduportal) [reply tests/fixtures/sample-with-diagram.adoc:55]: My apologies: I did not check the comment https://github.com/asciidoctor/docker-asciidoctor/pull/372#issuecomment-1613446738 carefully.

The fact you're updating this sample implies the "nominal" generation case is proving that a2s is used and you even demonstrated it in the command 👍

**Review by dduportal — APPROVED** [review]
Nice job, thanks for this contribution!

**Pull request merged**
